service/src/main/java/bio/terra/profile/app/common/MetricUtils.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.profile.app.common;
+
+import bio.terra.profile.model.CloudPlatform;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import io.micrometer.core.instrument.Metrics;
+import java.util.concurrent.Callable;
+
+public class MetricUtils {
+
+  private static final String CLOUD_PLATFORM_TAG = "cloudPlatform";
+
+  /**
+   * Emit a metric for the duration of creating a billing profile.
+   *
+   * @param callable the code to execute to create the profile, which will be timed
+   * @param platform the platform the profile will be created on
+   */
+  public static BillingProfile recordProfileCreation(
+      Callable<BillingProfile> callable, CloudPlatform platform) {
+    try {
+      return Metrics.globalRegistry
+          .timer("profile.creation.time", CLOUD_PLATFORM_TAG, platform.toString())
+          .recordCallable(callable);
+    } catch (Exception ex) { // Any exception bubbling up from the callable.
+      throw new RuntimeException(ex);
```

Comment:
The `recordCallable` method declares that it throws `Exception` (to bubble up anything from the callable). The actual method we are calling (`submitAndWait`) does not declare that it throws an Exception, so I'm converting any caught exception back into a RuntimeException. Do you think this is ok @aherbst-broad ?

This is how a re-thrown exception will appear in the logs:

```
{"timestampSeconds":1670533045,"timestampNanos":227000000,"severity":"ERROR","message":"Exception caught by catchall hander\njava.lang.RuntimeException: bio.terra.profile.service.profile.exception.DuplicateTagException@43d6a997[causes=[],statusCode=409 CONFLICT]\n\tat bio.terra.profile.app.common.MetricUtils.recordProfileCreation(MetricUtils.java:25)\n\tat bio.terra.profile.service.profile.ProfileService.createProfile(ProfileService.java:65)\n\tat bio.terra.profile.app.controller.ProfileApiController.createProfile(ProfileApiController.java:44)\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:568)\n\tat org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)\n\tat org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)\n\tat org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)\n\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)\n\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)\n\tat org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)\n\tat org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1071)\n\tat org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:964)\n\tat org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)\n\tat org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:696)\n\tat org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:779)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat bio.terra.common.logging.RequestLoggingFilter.doFilter(RequestLoggingFilter.java:60)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat bio.terra.common.logging.RequestCacheFilter.doFilter(RequestCacheFilter.java:29)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat bio.terra.common.logging.RequestIdFilter.doFilter(RequestIdFilter.java:51)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat io.opencensus.contrib.http.servlet.OcHttpServletFilter.doFilter(OcHttpServletFilter.java:176)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)\n\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)\n\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)\n\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)\n\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)\n\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)\n\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)\n\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)\n\tat org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)\n\tat org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)\n\tat org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:893)\n\tat org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1789)\n\tat org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)\n\tat org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)\n\tat org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)\n\tat org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\nCaused by: bio.terra.profile.service.profile.exception.DuplicateTagException: MRG has existing tag [mrg_id = mrg-terra-dev-previ-20221207134951, existing tag = terra.billingProfileId, value = 3fa45f64-5717-4962-b3fc-2c963f66afa6]\n\tat java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n\tat java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)\n\tat java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)\n\tat bio.terra.profile.service.job.StairwayExceptionSerializer.deserialize(StairwayExceptionSerializer.java:130)\n\tat bio.terra.stairway.impl.FlightDao.makeFlightStateList(FlightDao.java:1022)\n\tat bio.terra.stairway.impl.FlightDao.getFlightStateInner(FlightDao.java:852)\n\tat bio.terra.stairway.impl.FlightDao.lambda$getFlightState$11(FlightDao.java:831)\n\tat bio.terra.stairway.impl.DbRetry.perform(DbRetry.java:88)\n\tat bio.terra.stairway.impl.DbRetry.retry(DbRetry.java:66)\n\tat bio.terra.stairway.impl.FlightDao.getFlightState(FlightDao.java:831)\n\tat bio.terra.stairway.impl.StairwayImpl.getFlightState(StairwayImpl.java:530)\n\tat bio.terra.profile.service.job.JobService.retrieveJobResultWorker(JobService.java:353)\n\tat bio.terra.profile.service.job.JobService.retrieveJobResult(JobService.java:309)\n\tat bio.terra.profile.service.job.JobService.submitAndWait(JobService.java:125)\n\tat bio.terra.profile.service.job.JobBuilder.submitAndWait(JobBuilder.java:91)\n\tat bio.terra.profile.service.profile.ProfileService.lambda$createProfile$0(ProfileService.java:64)\n\tat io.micrometer.core.instrument.composite.CompositeTimer.recordCallable(CompositeTimer.java:77)\n\tat bio.terra.profile.app.common.MetricUtils.recordProfileCreation(MetricUtils.java:23)\n\t... 68 common frames omitted\n","serviceContext":{"service":"profile","version":"0.1.68-SNAPSHOT"},"context":"default","thread":"http-nio-8080-exec-1","logger":"bio.terra.common.exception.AbstractGlobalExceptionHandler","logging.googleapis.com/sourceLocation":{"file":"bio/terra/common/exception/AbstractGlobalExceptionHandler.java","line":52,"function":"bio.terra.common.exception.AbstractGlobalExceptionHandler.catchallHandler"},"logging.googleapis.com/trace":"projects/broad-dsde-dev/traces/36d3163a829a61fef4590fae8ae82e21","logging.googleapis.com/spanId":"265ae01785c294d0","logging.googleapis.com/trace_sampled":false,"requestId":"eO5DVOgD"}